### PR TITLE
Use non identity adapters in the use_adapter tests

### PR DESCRIPTION
**Pull request by albertvillanova**
The three `TestUseAdapter` tests pass no matter what `use_adapter` does.

They load `trl-internal-testing/tiny-PeftModel`, built by `peft_qwen3_for_causal_lm.py` with a default `LoraConfig()`. That zero-initializes `lora_B`, so the adapter starts as the identity, which is what you want for training and not for a fixture. I downloaded both published adapters to check: all four `lora_A` tensors are non-zero in each, and all four `lora_B` tensors are zero in both.

So the adapter contributes nothing, every state gives the base model's logits, and each assertion would hold with `use_adapter` replaced by a no-op context manager:

- `test_disables_on_none` compares disabled logits against `use_adapter(model, None)`, but enabled and disabled are already equal
- `test_restores_previous_adapter` checks the active adapter is restored, but every adapter state gives the same logits
- `test_with_multiple_adapters` loads a second adapter from `tiny-PeftModel-2`, whose whole reason to exist is having different weights, but both are the identity so `expected_1 == expected_2`

### Solution

Fix the fixtures rather than the tests: build both with `init_lora_weights=False`, which skips the zeroing. The tests keep loading them through `AutoPeftModelForCausalLM.from_pretrained`, and gain the two assertions they silently depended on, that the adapter changes the logits and that the two adapters differ.

Both Hub repos need regenerating for this to take effect, so the new assertions fail until then.

<details>
<summary>Superseded first approach</summary>

The first version rebuilt the adapters inside the test with `init_lora_weights=False` instead of touching the fixtures. It worked, but it dropped the suite's only use of `AutoPeftModelForCausalLM.from_pretrained`, which is the pattern `use_adapter`'s own docstring documents, and it left `tiny-PeftModel` and `tiny-PeftModel-2` with no consumer despite both having generation scripts. It also needed an explicit `target_modules` to avoid depending on peft's Qwen3 default mapping, added in peft 0.16.0 against a declared floor of 0.13.0.

</details>

---

> [!NOTE]
> **Low Risk**
> Test and tiny-model generation script changes only; runtime library behavior is unchanged aside from needing Hub fixture republish.
> 
> **Overview**
> **Fixes `TestUseAdapter` so it can actually fail** when `use_adapter` is broken. The Hub tiny PEFT fixtures were built with default `LoraConfig()`, which zeroes `lora_B` and makes every adapter an identity—enabled, disabled, and multiple adapters all produced identical logits.
> 
> Both `peft_qwen3_for_causal_lm.py` and `peft_qwen3_for_causal_lm_2.py` now use `LoraConfig(init_lora_weights=False)` so published `tiny-PeftModel` / `tiny-PeftModel-2` carry non-identity LoRA weights. **`TestUseAdapter`** adds pre-checks that adapter-on vs adapter-off logits differ and that the two adapters disagree, so the existing equality assertions are meaningful.
> 
> CI will stay red until those Hub repos are regenerated and pushed.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 0eacb4bd11f90a5d0c2a7d261f9dc819f3906edc. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -60,28 +60,39 @@
...
+        """Build a PEFT model whose adapters actually change the logits.
+
+        `LoraConfig()` leaves `lora_B` zeroed, which makes every adapter the identity: enabled, disabled and restored
+        all give the base model's logits, and the assertions below would then hold whatever `use_adapter` did.
+        `init_lora_weights=False` gives `lora_B` random values instead.
+        """
+        model = AutoModelForCausalLM.from_pretrained("trl-internal-testing/tiny-Qwen3ForCausalLM")
+        model = get_peft_model(model, LoraConfig(init_lora_weights=False), adapter_name=adapter_names[0])
+        for adapter_name in adapter_names[1:]:
+            model.add_adapter(adapter_name, LoraConfig(init_lora_weights=False))
```
[the remark this answers is not in the document]
Comment (albertvillanova) [reply tests/test_utils.py]: Correct, and specifically new to this PR. Fixed in 96054c5.

Verified both halves:

- peft added Qwen3 to its default LoRA target mapping in huggingface/peft#2522 (2025-04-29), first released in **v0.16.0**. Without it, `_prepare_adapter_config` hits `raise ValueError("Please specify `target_modules` or `target_parameters` in `peft_config`")`.
- TRL declares `peft>=0.13.0` in `pyproject.toml`.

And the reason this is new rather than pre-existing: every other bare `LoraConfig()` in the suite is applied to `tiny-Qwen2ForCausalLM-2.5`, and `qwen2` has been in that mapping far longer. Mine would have been the first test to need the Qwen3 entry. The old tests avoided it exactly as you say, by loading adapters whose `adapter_config.json` already named the targets.

Rather than raise the floor, the config now names the targets: `target_modules=["q_proj", "v_proj"]`, which is what peft's mapping selects for Qwen3 anyway, so the test no longer depends on any peft version for this.

Worth noting the gap that hid it: the minimum-versions CI job pins `accelerate`, `datasets` and `transformers` to their floors but not `peft`, so `peft>=0.13.0` is never exercised and CI would have stayed green on this either way.

**Comment by albertvillanova** [comment]
Holding this one for a moment. Rebuilding the adapters inside the test drops the only place the suite exercises `AutoPeftModelForCausalLM.from_pretrained`, which is the pattern `use_adapter`'s own docstring documents, and it leaves `tiny-PeftModel` and `tiny-PeftModel-2` without a consumer even though both have generation scripts.

The alternative is to fix the fixtures rather than route around them: build them with `init_lora_weights=False` in the generation scripts, regenerate both repos, and keep the tests loading them. An identity adapter is a defect in the fixture, not in how the test loads it.

Testing that the whole chain works before rewriting the PR around it, rather than proposing it on reasoning alone. Will report back.

**Comment by albertvillanova** [comment]
Changed approach. The earlier version rebuilt the adapters inside the test, which dropped the suite's only use of `AutoPeftModelForCausalLM.from_pretrained` and left both fixtures without a consumer. The identity adapter is a defect in the fixture, so it is fixed there instead. Verified the chain end to end before rewriting: with `init_lora_weights=False` the reloaded adapter changes the logits, two seeds give two different adapters, and the same seed reproduces one.

File: tests/test_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (albertvillanova) [reply tests/test_utils.py]: Moot now: the test no longer builds a `LoraConfig`, so there is no `target_modules` to set and no dependency on peft's Qwen3 default mapping. The fixtures are fixed at their source instead.

**Comment by albertvillanova** [comment]
The CI models have been regenerated and the corresponding Hub PRs are open. They need to be merged once this PR is approved:

- [x] https://huggingface.co/trl-internal-testing/tiny-PeftModel/discussions/3
- [x] https://huggingface.co/trl-internal-testing/tiny-PeftModel-2/discussions/2

**Comment by albertvillanova** [comment]
Tested with `MODEL_REVISIONS` pointing at the regenerated models, `trl-internal-testing/tiny-PeftModel` at `refs/pr/3` and `trl-internal-testing/tiny-PeftModel-2` at `refs/pr/2`, on top of this PR with #7288 cherry-picked: https://github.com/huggingface/trl/actions/runs/35353613892

The suite is green.

**Comment by albertvillanova** [comment]
Merging this PR as a CI fix.
